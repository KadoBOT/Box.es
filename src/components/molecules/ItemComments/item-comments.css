.item-comments {
  flex: 2;
  overflow: auto;
  position: relative;
  margin-top: 20px;
}

.item-comments__comment {
  display: flex;
  padding-top: 5px;
  padding-bottom: 5px;
}

.item-comments__load-more {
  margin-bottom: 10px;
  color: rgb(199, 199, 199);
  font-weight: 700;
  position: fixed;
  background: white;
  padding-bottom: 10px;
  width: 28%;
  z-index: 99;
  margin-top: -25px;
}

.item-comments__comment__text {
  padding: 10px 0px 0px 10px;
}

.item-comments__comment__text__author {
  color: black;
  font-weight: bold;
}
