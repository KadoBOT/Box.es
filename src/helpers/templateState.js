export default {
  "isFetching": false,
  "name": "Diamond Select Marvel Select: Juggernaut Action Figure",
  "description": "A Diamond Select Release! Sculpted by Gentle Giant! The ranks of the best-selling Marvel Select line expand to include one of the most deadly mutants ever, the unstoppable Juggernaut. Sculpted by Gentle Giant and standing a massive 7\" tall, Juggernaut is the latest in the, in scale, over sized figures from the Marvel Select line. Charles Xavier's half brother is ready to join Magneto in the fight against the X-Men and anyone else that stands in their way. Featuring Marvel Selects standard 16 points of articulation, Juggernaut, and the sans-helmet chase variant figure, are the textbook definition of Unstoppable!",
  "price": 22.43,
  "location_details": "Genosha",
  "currency": {
    "symbol": "$",
  },
  "likes_count": 80,
  "user": {
    "nickname": "xavier",
    "name": "Charles Xavier",
    "avatar": {
      "small_url": "http://static.comicvine.com/uploads/original/7/71941/1489898-charles_xavier.jpg",
    },
  },
  "item_photos": [
    {
      "large": "https://images-na.ssl-images-amazon.com/images/I/719022GksTL._SL1500_.jpg",
      "small": "https://images-na.ssl-images-amazon.com/images/I/719022GksTL._SL1500_.jpg",
      "id": 473580,
      "is_main": false
    },
    {
      "large": "http://www.angelfire.com/mech/jinsaotome/Blog/Juggyreview_001.JPG",
      "small": "http://www.angelfire.com/mech/jinsaotome/Blog/Juggyreview_001.JPG",
      "id": 473579,
      "is_main": false
    },
    {
      "large": "https://actionfigurecanada.files.wordpress.com/2011/09/juggernaut-marvel-select-figure-2.jpg",
      "small": "https://actionfigurecanada.files.wordpress.com/2011/09/juggernaut-marvel-select-figure-2.jpg",
      "id": 473581,
      "is_main": true
    },
    {
      "large": "http://4.bp.blogspot.com/-qNdk8wZtZSY/UgGmCHgBLJI/AAAAAAAB0pk/2qpwZTuNSls/s1600/52458-1.jpg",
      "small": "http://4.bp.blogspot.com/-qNdk8wZtZSY/UgGmCHgBLJI/AAAAAAAB0pk/2qpwZTuNSls/s1600/52458-1.jpg",
      "id": 473582,
      "is_main": false
    },
    {
      "large": "https://s-media-cache-ak0.pinimg.com/originals/08/38/8f/08388f9c134afd90f567be95a4fe610b.jpg",
      "small": "https://s-media-cache-ak0.pinimg.com/originals/08/38/8f/08388f9c134afd90f567be95a4fe610b.jpg",
      "id": 473583,
      "is_main": false
    }
  ],
  "recent_comments": [
    {
      "id": 184543,
      "comment": "Miss those classic outfits ",
      "user": {
        "id": 13118,
        "nickname": "colossus",
        "name": "Piotr Rasputin",
        "avatar": {
          "small_url": "http://cdn2-www.comingsoon.net/assets/uploads/2015/12/deadpool-colossus-header.jpg",
          "is_default": false
        },
        "share_url": "http://box.es/nixon"
      }
    },
  ],
}
