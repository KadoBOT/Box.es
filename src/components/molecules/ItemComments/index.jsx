import React, {PropTypes} from 'react'

import { Avatar, Description } from '../../atoms/'
import './item-comments.css'

const Comments = ({comments}) => (
  <div className="item-comments">
    <p className="item-comments__load-more">load more comments</p>
    {comments[0] === 'isFetching' && <div>Loading Comments...</div>}
    {comments[0] !== 'isFetching' && comments.map(c => (
      <div
        className="item-comments__comment"
        key={c.id}
      >
        <a href={c.user.share_url}>
          <Avatar img={c.user.avatar.small_url} />
        </a>
        <div className="item-comments__comment__text">
          <Description comment>
            <a href={c.user.share_url}>
              <span className="item-comments__comment__text__author">{c.user.nickname}</span>
            </a>
            {' '}
            <span title={c.comment}>{c.comment}</span>
          </Description>
        </div>
      </div>
    ))}
  </div>
)

Comments.propTypes = {
  comments: PropTypes.array.isRequired
}

export default Comments
