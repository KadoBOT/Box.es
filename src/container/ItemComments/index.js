import { connect } from 'react-redux';
import R from 'ramda'

import ItemComments from '../../components/molecules/ItemComments'

const mapStateToProps = R.pick(['comments'])

export default connect(mapStateToProps)(ItemComments);
