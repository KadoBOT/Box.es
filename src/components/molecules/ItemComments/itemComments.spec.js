import React from 'react'
import renderer from 'react-test-renderer';

import ItemComments from './index'
import templateState from '../../../helpers/templateState'

it('should render the ItemComments', () => {
  const wrapper = renderer.create(
      <ItemComments comments={templateState.comments} />
  );

  expect(wrapper.toJSON()).toMatchSnapshot();
  templateState.comments = ['isFetching']

});

it('should render the ItemComments', () => {
  const wrapper = renderer.create(
      <ItemComments comments={templateState.comments} />
  );

  expect(wrapper.toJSON()).toMatchSnapshot();
});
