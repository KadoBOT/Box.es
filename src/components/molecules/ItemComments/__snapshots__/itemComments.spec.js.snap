exports[`test should render the ItemComments 1`] = `
<div
  className="item-comments">
  <p
    className="item-comments__load-more">
    load more comments
  </p>
  <div
    className="item-comments__comment">
    <a
      href="http://box.es/colossus">
      <div
        className="avatar"
        style={
          Object {
            "background": "url(http://cdn2-www.comingsoon.net/assets/uploads/2015/12/deadpool-colossus-header.jpg)",
          }
        } />
    </a>
    <div
      className="item-comments__comment__text">
      <div
        className="fTcZCa">
        <a
          href="http://box.es/colossus">
          <span
            className="item-comments__comment__text__author">
            colossus
          </span>
        </a>
         
        <span
          title="We will make an X-Man of you yet, Wade">
          We will make an X-Man of you yet, Wade
        </span>
      </div>
    </div>
  </div>
  <div
    className="item-comments__comment">
    <a
      href="http://box.es/deadpool">
      <div
        className="avatar"
        style={
          Object {
            "background": "url(http://nerdist.com/wp-content/uploads/2016/10/20161025_n_nerdistnews_deadpool2director_1x1.jpg)",
          }
        } />
    </a>
    <div
      className="item-comments__comment__text">
      <div
        className="fTcZCa">
        <a
          href="http://box.es/deadpool">
          <span
            className="item-comments__comment__text__author">
            deadpool
          </span>
        </a>
         
        <span
          title="If I ever decide to become a crime-fighting shit swizzler, who rooms with a bunch of other little whiners at Neverland Mansion with some creepy, old, bald, Heaven\'s Gate-looking motherfucker... on that day, I\'ll send your shiny, happy ass a friend request!">
          If I ever decide to become a crime-fighting shit swizzler, who rooms with a bunch of other little whiners at Neverland Mansion with some creepy, old, bald, Heaven\'s Gate-looking motherfucker... on that day, I\'ll send your shiny, happy ass a friend request!
        </span>
      </div>
    </div>
  </div>
  <div
    className="item-comments__comment">
    <a
      href="http://box.es/ntw">
      <div
        className="avatar"
        style={
          Object {
            "background": "url(https://s-media-cache-ak0.pinimg.com/736x/48/c9/8e/48c98e3d4dbc0fc782cb1e6877b59a33.jpg)",
          }
        } />
    </a>
    <div
      className="item-comments__comment__text">
      <div
        className="fTcZCa">
        <a
          href="http://box.es/ntw">
          <span
            className="item-comments__comment__text__author">
            negasonic teenage warhead
          </span>
        </a>
         
        <span
          title="You guys going for a bite? Early bird special?">
          You guys going for a bite? Early bird special?
        </span>
      </div>
    </div>
  </div>
  <div
    className="item-comments__comment">
    <a
      href="http://box.es/blindal">
      <div
        className="avatar"
        style={
          Object {
            "background": "url(http://starkindustries.ru/_dr/2/60896654.png)",
          }
        } />
    </a>
    <div
      className="item-comments__comment__text">
      <div
        className="fTcZCa">
        <a
          href="http://box.es/blindal">
          <span
            className="item-comments__comment__text__author">
            blind al
          </span>
        </a>
         
        <span
          title="Love is blind, Wade">
          Love is blind, Wade
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`test should render the ItemComments 2`] = `
<div
  className="item-comments">
  <p
    className="item-comments__load-more">
    load more comments
  </p>
  <div>
    Loading Comments...
  </div>
</div>
`;
